File: lua/p1.lua
function sum_divisible(n, max)
  p = max // n
  return n * (p * (p + 1)) // 2
end

max = 999
sum = sum_divisible(3, max) + sum_divisible(5, max) - sum_divisible(15, max)
print(sum)

File: go/p1.go
package main

import "fmt"
import "os"
import "strconv"

func SumDivisible(n, max int) int {
	var p int = max / n
	return n * (p * (p + 1)) / 2
}

func main() {
	var max int = 999
	if len(os.Args) > 1 {
		if s, err := strconv.Atoi(os.Args[1]); err == nil {
			max = s - 1
		}
	}
	var sum int = SumDivisible(3, max) + SumDivisible(5, max) - SumDivisible(15, max)
	fmt.Printf("%v\n", sum)
}
